### PR TITLE
Deprecate peeling wetting

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- The peeling and wetting mechanism was very ad-hoc and not being used anymore. It was also not robust because of all the changes made to streamline the VOF solver. This is a right time to deprecate it and, if need be in the future or if we find people needing a similar mechanism, we can re-introduce it more robustly (using the compressible equation of state with it for example)

# Description of the solution

- Deprecate

# How Has This Been Tested?

- peeling-wetting tests have been removed

# Documentation

- Peeling-wetting documentation has been removed

# Comments

- It's cool to deprecate stuff from time to time. I think it's a healthy reflection on the state of the code
- I will deprecate the "conservative fluid" stuff next

### CODE REVIEW

**Review by hepap — COMMENTED** [review]
There are comments mentioning the wetting/peeling
`navier_stokes_vof_assemblers.cc`, lines 13,244,677,908
`vof_assemblers.cc` lines 29,167
`vof_scratch_data.h` line 232

File: include/core/boundary_conditions.h [hunk 1]
```diff
@@ -974,9 +970,9 @@ namespace BoundaryConditions
   {
     prm.declare_entry("type",
                       "none",
-                      Patterns::Selection("none|dirichlet|peeling/wetting"),
+                      Patterns::Selection("none|dirichlet"),
                       "Type of boundary condition for VOF"
-                      "Choices are <none|dirichlet|peeling/wetting>.");
+                      "Choices are <none|dirichlet.");
```
Comment (hepap) [inline include/core/boundary_conditions.h:975]:
```suggestion
                      "Choices are <none|dirichlet>.");
```

File: source/solvers/vof_scratch_data.cc [hunk 2]
```diff
@@ -47,36 +47,6 @@ VOFScratchData<dim>::allocate()
...
-{
-  switch (properties_manager.get_number_of_fluids())
-    {
-        case 1: {
-          // Cannot proceed a two-phase flow simulations if only one fluid is
-          // defined. This error must have been thrown a long time ago by the
-          // code, it is repeated here for clarity.
-          throw std::runtime_error(
-            "Cannot run VOF simulations with only 1 fluid, review the Physical Properties section of the .prm");
-        }
```
Comment (hepap) [inline source/solvers/vof_scratch_data.cc:68]: Do we have this check somewhere else? I didn't find it :o

**Review by hepap — COMMENTED** [review]
Only two things left I think :)

File: source/solvers/vof_scratch_data.cc (hunk 2 above)
Comment (blaisb) [reply source/solvers/vof_scratch_data.cc:68]: Nope, that's a good point, but it also should not be there in the scratch data. I will add the assertion in the vof constructor, that's where it belongs

**Review by hepap — APPROVED** [review]

**Review by AmishgaAlphonius — APPROVED** [review]
Two last changes :)

File: include/solvers/vof.h
Comment (AmishgaAlphonius) [inline include/solvers/vof.h:1]: line 243 mentions ``wetting/peeling``

**Pull request merged**